configure.ac

```diff
@@ -107,20 +107,23 @@ KVDB_CHECK_BUILTIN([__sync_lock_release_set],
 
 dnl C++ features
 
-AC_CACHE_CHECK([whether the C++ compiler understands 'auto'], [ac_cv_cxx_auto], [
+dnl Mimics AC_CACHE_CHECK(MESSAGE, CACHE-ID, COMMANDS), but retries COMMANDS
+dnl with CXX='${CXX} -std=gnu++0x' if the user has not specified CXX.
+AC_DEFUN([KVDB_CHECK_C11_FEATURE], [
+    AC_CACHE_CHECK([$1], [$2], [$3])
+
+    if test "$$2" != yes -a -z "$ac_user_cxx"; then
+        CXX="${CXX} -std=gnu++0x"
+        AC_MSG_CHECKING([$1 with -std=gnu++0x])
+        $3
+        AC_MSG_RESULT([$$2])
+    fi
+])
+
+KVDB_CHECK_C11_FEATURE([whether the C++ compiler understands 'auto'], [ac_cv_cxx_auto], [
     AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[struct s { int a; }; int f(s x) { auto &y = x; return y.a; }]], [[]])],
 	[ac_cv_cxx_auto=yes], [ac_cv_cxx_auto=no])])
-if test "$ac_cv_cxx_auto" != yes -a -z "$ac_user_cxx"; then
-    CXX="${CXX} -std=gnu++0x"
-    AC_MSG_CHECKING([whether the C++ compiler with -std=gnu++0x understands 'auto'])
-    AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[struct s { int a; }; int f(s x) { auto &y = x; return y.a; }]], [[]])],
-	[ac_cv_cxx_auto=yes], [ac_cv_cxx_auto=no])
-    AC_MSG_RESULT([$ac_cv_cxx_auto])
-fi
-
-if test "$ac_cv_cxx_auto" = yes; then
-    AC_DEFINE([HAVE_CXX_AUTO], [1], [Define if the C++ compiler understands 'auto'.])
```

Comment:
This define seemed rather silly, since we abort the build if we don't have `auto`.